### PR TITLE
Fixes & improvements

### PR DESCRIPTION
**Pull request by jploskonka**
- Refactor code to more ”awk way”
- Handle `HEAD "/"` requests (used by monitoring services to check if site is up) separately from group stats.
- Correctly handle requests with space in parameter value
- Add hostnames report

### CODE REVIEW

**Comment by oma** [comment]
I like that you invest in a good PR description, using bullets 👍

File: report [hunk 1]
```diff
@@ -17,72 +17,58 @@ fi
...
 
-    printf("\n%s\n","***REMOVED***" )
-    i = 1; for(k in STATUS_500) {
-      print STATUS_500[k] ": " k
-
-      i++; if ( i == 10 ) { break }
-    }
+  match($0, /status=([0-9]+)/, status)   { STATUS[status[1]]++ }
+  match($0, /status=500/)                { STATUS_500[_path]++ }
+  match(_path, /\/(assets\/|favicon\.)/) { ASSET_REQ[_path]++ }
```
Comment (oma) [inline report]: much nicer ! 👍

File: README.md [hunk 2]
```diff
@@ -28,10 +28,5 @@ gitignored so don't worry about accidentally adding log file to git repo. After
 getting file there's simple report generated with awk.
 
 ## Known issues
-- newrelic (and probably honeybadger) performs `HEAD /` requests to check if
-  site is up. This affects number of requests with `status=200` and `path="/"`.
-  It would be probably best to just filter those requests out from stats.
 - some requests are marked as failed by heroku (those with `code=XX`) and are
   filtered out right now. It's `~3000-4000` requests per day.
```
Comment (oma) [inline README.md]: I'd like some stats on those as well. Wether they are H14 out of memory or other :)

Comment (oma) [inline README.md]: give me a finger and I'll take the whole hand :D

**Comment by oma** [comment]
on the topic of 'grabbing the whole hand', it'd be interested knowing more about what the top requesters are actually requesting. top 10 request og top 10 IPs :)
Up to you. btw, you rock

File: README.md (hunk 2 above)
Comment (jploskonka) [inline README.md]: Yeah those H=XX requests are definitely thing to handle, maybe next week :)

**Pull request merged**